src/storage/object.ts

```diff
@@ -584,19 +591,21 @@ export class ObjectStorage {
     }
 
     // check if user has INSERT permissions
-    await this.db.testPermission((db) => {
-      return db.createObject({
-        bucket_id: this.bucketId,
-        name: objectName,
-        owner,
-        metadata: {},
-      })
+    await this.uploader.canUpload({
+      bucketId: this.bucketId,
+      objectName,
+      owner,
+      isUpsert: options?.upsert || false,
```

Comment:
```suggestion
      isUpsert: options?.upsert ?? false,
```
in this case doesn't matter cos default value is false, but in general good to use ?? for checking for existence of a property